packages/ethereum-swap-provider/lib/EthereumSwapProvider.js

```diff
@@ -70,31 +70,48 @@ export default class EthereumSwapProvider extends Provider {
       '60', '48', // PUSH1 48
       'f1', // CALL
 
+      // Validate input size
+      '36', // CALLDATASIZE
+      '60', '20', // PUSH1 20 (32 bytes)
+      '14', // EQ
+      '16', // AND (to make sure SHA256 CALL succeeded)
+
       // Validate with secretHash
       '7f', secretHash, // PUSH32 {secretHashEncoded}
       '60', '21', // PUSH1 21
       '51', // MLOAD
       '14', // EQ
-      '16', // AND (to make sure CALL succeeded)
+      '16', // AND (to make sure input valid size)
       // Redeem if secret is valid
-      '60', '47', // PUSH1 {redeemDestination}
+      '60', '4c', // PUSH1 {redeemDestination}
       '57', // JUMPI
 
       // Check time lock
       '64', expirationEncoded, // PUSH5 {expirationEncoded}
       '42', // TIMESTAMP
       '11', // GT
       // Refund if timelock passed
-      '60', '5e', // PUSH1 {refundDestination}
+      '60', '89', // PUSH1 {refundDestination}
       '57',
 
       'fe', // INVALID
 
       '5b', // JUMPDEST
+      // emit Claim(bytes32 _secret)
+      '7f', '8c1d64e3bd87387709175b9ef4e7a1d7a8364559fc0e2ad9d77953909a0d1eb3', // PUSH32 topic Keccak-256(Refund())
+      '60', '20', // PUSH1 20 (log length - 32)
+      '60', '00', // PUSH1 00 (log offset - 0)
+      'a1', // LOG 1
       '73', recipientAddress, // PUSH20 {recipientAddressEncoded}
       'ff', // SELF-DESTRUCT
 
       '5b', // JUMPDEST
+      // emit Refund()
+      '7f', '5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed55178', // PUSH32 topic Keccak-256(Refund())
+      '60', '00', // PUSH1 00 (log length - 0)
+      '80', // DUP 1 (log offset)
+      'a1', // LOG 1
```

Comment:
`event Refund ()` doesn't need any topics. can we use `a0`?

Reply:
The first topic of a log is the event name signature

Reply:
yes, but can we use `LOG0` opcode instead of `LOG1`?

```
'7f', '5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed55178', // PUSH32 topic Keccak-256(Refund())
'60', '00', // PUSH1 00 (log length - 0)
// '80', // DUP 1 (log offset) // remove DUP 1
'a0' // change a1 to a0
```

Reply:
LOG0 is data without topic
We need topic i.e. the refund event no?
Try to compile the event in remix, you will see log1 and topic is kecakk-256(refund()) same behaviour here